Execution API (#396)
grafana/k6-docs: src/data/markdown/docs/02 javascript api/06 k6-execution.md

---
title: "k6/execution"
excerpt: "Get information about the current test's execution state."
---

k6 v0.34.0 introduced the capability for getting information about the current test execution state, from inside of the test's script.

It can be used by importing the `k6/execution` module.

<div class="code-group" data-props='{"labels": [], "lineNumbers": [true]}'>

```javascript
import exec from 'k6/execution';

var options = {

> 	scenarios: {
> 		'myscenario': { // this will be the returned name
> 			executor: 'shared-iterations',
> 			maxDuration: '30m'
> 		}
> 	}
> }
> 
> export default function() {
> 	console.log(exec.scenario.name) // myscenario 

Reviewer: Two-space indents instead of tabs ;)

I would also change the other examples.

```suggestion
  scenarios: {
    'myscenario': { // this will be the returned name
      executor: 'shared-iterations',
      maxDuration: '30m'
    }
  }
}

export default function() {
  console.log(exec.scenario.name) // myscenario 
```

}
```

</div>

> The module has a default export that has the following keys: `scenario`, `instance` and `vu` which in themselves expose information specific to the context in which it's being accessed.

Reviewer 2: ```suggestion
The module has a default export that has the following keys: `scenario`, `instance` and `vu`, which in themselves expose information specific to the context in which they are being accessed.
```

> ### ℹ️ Identifiers
>

> > All unique identifiers are sequentially generated starting from a base of zero or one. Occasionally, in rare circumstances, the sequence could skip some values.

Reviewer 3: ```suggestion
> All unique identifiers are sequentially generated starting from a base of zero(iterations) or one(vu ids). Occasionally, in rare circumstances, the sequence could skip some values.
```

> > All unique identifiers are sequentially generated starting from a base of zero (iterations) or one (vu ids). Occasionally, in rare circumstances, the sequence could skip some values.

Reviewer 2: ```suggestion
> All unique identifiers are sequentially generated starting from a base of zero (iterations) or one (vu ids). In distributed/cloud test runs, the test-wide iteration numbers and VU identifiers are still going to be unique across instances, though there might be gaps in the sequences when, for example, some instances execute faster iterations than others or allocate more VUs mid-test.
```

### Scenario
| Field               | Type    | Description                                                              |
|---------------------|---------|--------------------------------------------------------------------------|

> | name                | string  | It is the assigned name of the running scenario.                                |
> | executor            | string  | It is the name of the running [Executor](https://k6.io/docs/using-k6/scenarios/#executors) type.                                                |
> | startTime           | integer | It represents the unix timestamp in milliseconds when the scenario started.                                                                           |

Reviewer: ```suggestion
| name                | string  | The assigned name of the running scenario.                                |
| executor            | string  | The name of the running [Executor](https://k6.io/docs/using-k6/scenarios/#executors) type.                                                |
| startTime           | integer | The Unix timestamp in milliseconds when the scenario started.                                                                           |
```

| progress            | float   | Percentage in a 0 to 1 interval of the scenario progress.                    |
| iterationInInstance | integer | The unique and zero-based sequential number of the current iteration in the scenario, across the current instance. |

> | iterationInTest     | integer | The unique and zero-based sequential number of the current iteration in the scenario, across the entire test. It is unique in all k6 execution modes - in local, cloud and distributed/segmented test runs. However, with cloud/distributed tests, every instance will get non-overlapping index values, but they might iterate over them at different speeds, so the values won't be sequential across them. |

Reviewer 2: ```suggestion
| iterationInTest     | integer | The unique and zero-based sequential number of the current iteration in the scenario, across the entire test. It is unique in all k6 execution modes - in local, cloud and distributed/segmented test runs. However, while every instance will get non-overlapping index values in cloud/distributed tests, they might iterate over them at different speeds, so the values won't be sequential across them. |
```

| iterationsCompleted     | integer | The number of completed iterations in the current instance.  |
| vusActive            | integer  | The number of active VUs.                                                |
| vusInitialized           | integer | The number of currently initialized VUs.                                                                           |
| currentTestRunDuration            | float   | The time passed from the start of the current test run in milliseconds.                    |

### VU
| Field               | Type    | Description                                                              |
|---------------------|---------|--------------------------------------------------------------------------|
| iterationInInstance | integer | The identifier of the iteration in the current instance.                                                                 |
| iterationInScenario | integer | The identifier of the iteration in the current scenario.                  |
| idInInstance        | integer | The identifier of the VU across the instance.                            |

> | idInTest            | integer | The global-unique identifier of the VU.                                  |

Reviewer 2: ```suggestion
| idInTest            | integer | The globally unique (across the whole test run) identifier of the VU.                                  |
```
